src/worker/generate-zip.ts

```diff
@@ -16,10 +16,50 @@ type FileList = { [filename: string]: string };
  *  })
  */
 export function createZipWithFiles(files: FileList): Promise<ArrayBuffer> {
-  const zip = new JSZip();
+  const zip = createJSZip();
   for (const [filename, contents] of Object.entries(files)) {
     zip.file(filename, contents);
   }
 
   return zip.generateAsync({ type: "arraybuffer" });
 }
+
+/**
+ * This pile of hacks tries to load the JSZip library, whether in Node.JS or
+ * in the WebWorker context.
+ *
+ * In the WebWorker context, it is assumed that JSZip was loaded globally, and
+ * is accessible as self.JSZip.
+ *
+ * In Node.JS, you require() it like any other module.
+ */
+function createJSZip(): JSZip {
+  let Constructor: typeof JSZip;
+
+  if (typeof self !== "undefined" && jsZipIsDefinedWithin(self)) {
+    // Within the WebWorker:
+    Constructor = self.JSZip;
+  } else {
+    // Within Node.js (Ava unit tests, probably)
+    Constructor = require("jszip");
+  }
+  return new Constructor();
+}
+
+type WorkerGlobalScopeWithJSZip = WorkerGlobalScope & { JSZip: typeof JSZip };
+
+/**
+ * This **type guard** returns true when the global scope seems to have the
+ * JSZip constructor defined.
+ *
+ * When used in an `if ()` condition, then the scope within the if-block
+ * **definitely** has a global called JSZip!
+ *
+ * See: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards
+ */
+function jsZipIsDefinedWithin(
+  scope: WorkerGlobalScope
+): scope is WorkerGlobalScopeWithJSZip {
+  const augmentedScope = scope as Partial<WorkerGlobalScopeWithJSZip>;
+  return typeof augmentedScope.JSZip === "function";
+}
```

Comment:
Hey @pranavnarang! I changed this code to (hopefully) be less mysterious than before. Let me know if this makes more sense!